stdlib/src/collections/dict.mojo

```diff
@@ -790,7 +790,16 @@ struct Dict[K: KeyElement, V: CollectionElement](
             var entry_value = entry[].unsafe_take()
             entry[] = None
             self.size -= 1
-            return entry_value.value^
+            var tmp = entry_value.value^ #TODO: not to merge with current PR
+            #It is necessary for def test_pop_default()
```

Comment:
Missing space after hash :)

Reply:
Why not just return `entry_value.value^` directly?

Reply:
:+1: Thanks @bethebunny , @gryznar, 

It is related to this issue: 
- https://github.com/modularml/mojo/pull/2796

The test `test_pop_default()` would fail,

I think it has to do with values that needs to be fully initialized to be ready for the `__del__` ,

but we move the value out anyway so maybe it has to do with something else ?